shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt

```diff
@@ -277,6 +277,35 @@ object JsonLdUtils {
         entities.map {
             it.compact()
         }
+
+    fun filterJsonldMapOnAttributes(input: Map<String, Any>, includedAttributes: Set<String>): Map<String, Any> {
+        return if (includedAttributes.isEmpty()) {
+            input
+        } else {
+            val mandatoryKeys = setOf("id", "type", "@context")
+            val includedKeys = mandatoryKeys.plus(includedAttributes)
+            input.filterKeys { includedKeys.contains(it) }
+        }
+    }
+
+    fun simplifyJsonldMap(input: Map<String, Any>): Map<String, Any> {
```

Comment:
Maybe you can create: 
```
Map<String, Any>.toKeyValueRepresentation = this.map { (key, value) -> 
    key to simplifyValue(value)
}.toMap()

```
And in the simplifyValue function you do the simplification needed.

Reply:
OK